framework/auth/core.py

```diff
@@ -946,18 +947,17 @@ def get_confirmation_url(self, email, external=True, force=False, renew=False, e
         :param force: If an expired token exists for the given email, generate a new one and return it.
         :param renew: Generate a new token and return it.
         :param external_id_provider: The external identity provider that authenticates the user.
+        :param service_url: The destination url to redirect after confirmation
         :return: Return the confirmation url.
         :raises: ExpiredTokenError if trying to access a token that is expired.
         :raises: KeyError if there is no token for the email.
         """
 
         base = settings.DOMAIN if external else '/'
         token = self.get_confirmation_token(email, force=force, renew=renew)
-
-        if external_id_provider:
-            return '{0}confirm/external/{1}/{2}/'.format(base, self._primary_key, token)
-        else:
-            return '{0}confirm/{1}/{2}/'.format(base, self._primary_key, token)
+        external = 'external/' if external_id_provider else ''
+        next = '?{}'.format(urllib.urlencode({'next': service_url})) if service_url else ''
```

Comment:
`next` is a reserved word, and just because Python lets you assign things to reserved words doesn't mean you should:

```
In [1]: True = 0

In [2]: bool(True)
Out[2]: False

In [3]: True == False
Out[3]: True
```

Prefer something like `destination` or `next_url`.